src/Missile.cpp

```diff
@@ -172,24 +172,51 @@ bool Missile::OnDamage(Object *attacker, float kgDamage, const CollisionContact
 	return true;
 }
 
+double calcAreaSphere(const double r) {
+	return 4.0*M_PI*r*r;
+}
+
+double calcAreaCircle(const double r) {
+	return M_PI*r*r;
+}
+
 void Missile::Explode()
 {
 	Pi::game->GetSpace()->KillBody(this);
 
-	const double damageRadius = 200.0;
-	const double kgDamage = 10000.0;
+	double kgYield = 200.0; // yield of the used explosive in kg
+	const double jEnergyPerKgExplosive = 4.184e6; // energy density of explosive, here it's TNT
+
+	const double jYield = kgYield * jEnergyPerKgExplosive; // how much energy was converted in the explosion?
 
+	double queryRadius = 2000.0; // hardcoded to 2 km, this is sufficient for most explosions
 	CollisionContact dummy;
-	Space::BodyNearList nearby = Pi::game->GetSpace()->GetBodiesMaybeNear(this, damageRadius);
+	Space::BodyNearList nearby = Pi::game->GetSpace()->GetBodiesMaybeNear(this, queryRadius);
 	for (Body *body : nearby) {
 		if (body->GetFrame() != GetFrame()) continue;
-		double dist = (body->GetPosition() - GetPosition()).Length();
-		if (dist < damageRadius) {
-			// linear damage decay with distance
-			body->OnDamage(m_owner, kgDamage * (damageRadius - dist) / damageRadius, dummy);
-			if (body->IsType(Object::SHIP))
-				LuaEvent::Queue("onShipHit", dynamic_cast<Ship *>(body), m_owner);
-		}
+		double dist = (body->GetPosition() - GetPosition()).Length(); // distance from explosion in meter
+
+		double targetRadius = 6.0; // radius of the hit target in meter
```

Comment:
You can get the radius of the body from `body->GetRadius()` I think

Reply:
thx, I'll use   it and retune the variable to get roughtly the same damage as the old code.